### PR TITLE
Fix type annotation on overloads

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
- Make type overloads use same way of generating parameter types as tooltip/hover. This way get get proper short types and not something like `a:T, int` on a generic or `a:list[float, float, float], list[float]`.
- Return value should not have `[ ]` around it if it is a single type.
- If function returns nothing, then display `None` and not `[ ]`
- Prettify documentation with proper spacing
- Update all TypingModuleDocumentation* tests baselines

### CODE REVIEW

File: src/Analysis/Engine/Impl/OverloadResult.cs [hunk 1]
```diff
@@ -117,44 +112,54 @@ public class OverloadResult : IOverloadResult {
...
             return new OverloadResult(parameters, name, doc, returnType);
         }
 
         public override string ToString() {
-            return "{0}({1})->[{2}]{3}".FormatInvariant(
-                Name,
-                string.Join(",", Parameters.Select(p => "{0}{1}:{2}={3}".FormatInvariant(p.Name, p.IsOptional ? "?" : "", p.Type ?? "", p.DefaultValue ?? ""))),
-                string.Join(",", ReturnType.OrderBy(k => k)),
-                string.IsNullOrEmpty(Documentation) ? "" : ("'''{0}'''".FormatInvariant(Documentation))
+            var parameters = string.Join(", ",
```
Comment (AlexanderSher) [inline src/Analysis/Engine/Impl/OverloadResult.cs]: Where do we use this `ToString` implementation? In serialization?

Comment (MikhailArkhipov) [reply src/Analysis/Engine/Impl/OverloadResult.cs]: It actually generates the signature

File: src/Analysis/Engine/Test/TypeAnnotationTests.cs [hunk 2]
```diff
@@ -392,13 +391,13 @@ private class StringConverter : TypeAnnotationConverter<string> {
 new_vector = scale(2.0, [1.0, -4.2, 5.4])
 ",
                 new[] {
-                    "scale:scale(scalar:float=,vector:list[float]=)->[list,list[float]]"
+                    "scale:scale(scalar:float, vector:list[float]) -> list[float]"
```
Comment (AlexanderSher) [inline src/Analysis/Engine/Test/TypeAnnotationTests.cs]: So `[list,list[float]]` is actually merged into `list[float]`?

File: src/Analysis/Engine/Impl/Values/BuiltinInstanceInfo.cs [hunk 3]
```diff
@@ -33,6 +33,7 @@ public BuiltinInstanceInfo(BuiltinClassInfo klass)
         IBuiltinClassInfo IBuiltinInstanceInfo.ClassInfo => ClassInfo;
         public BuiltinClassInfo ClassInfo => _klass;
 
+        public override string Name => _klass.Name;
```
Comment (AlexanderSher) [inline src/Analysis/Engine/Impl/Values/BuiltinInstanceInfo.cs]: Do we need similar change for `InstanceInfo`?

File: src/Analysis/Engine/Test/TypeAnnotationTests.cs (hunk 2 above)
Comment (MikhailArkhipov) [reply src/Analysis/Engine/Test/TypeAnnotationTests.cs]: Yes. Tooltips were correct while signatures were really weird

File: src/Analysis/Engine/Impl/OverloadResult.cs [hunk 4]
```diff
@@ -172,17 +177,7 @@ class AccumulatedOverloadResult {
             return x.Length >= y.Length ? x : y;
         }
 
-        private IAnalysisSet ChooseBest(IAnalysisSet x, IAnalysisSet y) {
```
[the remark this answers is not in the document]
Comment (MikhailArkhipov) [reply src/Analysis/Engine/Impl/OverloadResult.cs]: Yes

File: src/Analysis/Engine/Impl/Values/BuiltinInstanceInfo.cs (hunk 3 above)
Comment (MikhailArkhipov) [reply src/Analysis/Engine/Impl/Values/BuiltinInstanceInfo.cs]: Won't hurt, I'll add them.

**Pull request merged**